expression/src/main/java/io/smallrye/common/expression/Expression.java

```diff
@@ -269,16 +272,16 @@ private static Node parseString(Itr itr, final boolean allowExpr, final boolean
                         start = itr.getNextIdx();
                         continue;
                     }
-                    // enqueue what we have acquired so far
-                    if (idx > start) {
-                        // TP 4
-                        list.add(new LiteralNode(itr.getStr(), start, idx));
-                    }
```

Comment:
This can be undone since the same piece should be in every possible subcase now.